doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -946,13 +946,18 @@ described first.
 
 .. cmd:: Strategy {+ @strategy_level [ {+ @smart_qualid } ] }
 
-   .. insertprodn strategy_level strategy_level
+   .. insertprodn strategy_level strategy_level strategy_level_or_var
 
    .. prodn::
       strategy_level ::= opaque
       | @int
       | expand
       | transparent
+      strategy_level_or_var ::= @strategy_level
+      | @ident
+
+   .. todo pr is this the right location for strategy_level_or_var?
+      does it need more doc?  where does it belong?
```

Comment:
The location looks good to me. You might want to try to regenerate the grammar files `fullGrammar` and `orderedGrammar` following your changes. Then you might need to manually move the rules for `strategy_level_or_var` next to `strategy_level` (in `orderedGrammar`). Have a look at `doc/tools/doc_grammar/README.md`. No worries if you don't have time to do this (this is not checked by the CI and can be adjusted by Jim or I later on).

Reply:
I will leave this to you and Jim; when I run `make doc_gram; make doc_gram_rsts`, it *removes* `strategy_level_or_var` from the .rst file, adds a `| @strategy_level0` entry to `strategy_level` (which exists in the code to "work around a limitation of the macro system" (cc @ppedrot )), and I get a bunch of warnings:
```
DOC_GRAM_RSTS
# cmds in rsts, gram, total = 275 243 422
# tacs in rsts, gram, total = 267 0 267
Warning: editedGrammar: Undefined symbol 'strategy_level0'
Warning: editedGrammar: Unreachable symbol 'dangling_pattern_extension_rule'
Warning: editedGrammar: Unreachable symbol 'vernac_aux'
Warning: editedGrammar: Unreachable symbol 'subprf'
Warning: editedGrammar: Unreachable symbol 'finite_token'
Warning: editedGrammar: Unreachable symbol 'tactic_mode'
Warning: editedGrammar: Unreachable symbol 'ltac_use_default'
Warning: editedGrammar: Unreachable symbol 'tacticals'
Warning: Singleton non-terminal, maybe SPLICE?: first_letter
Warning: Singleton non-terminal, maybe SPLICE?: ltac_expr_opt
Warning: Singleton non-terminal, maybe SPLICE?: num
Warning: Singleton non-terminal, maybe SPLICE?: scope_key
Warning: Singleton non-terminal, maybe SPLICE?: scope_name
Warning: Singleton non-terminal, maybe SPLICE?: setting_name
Warning: Singleton non-terminal, maybe SPLICE?: subsequent_letter
Warning: Singleton non-terminal, maybe SPLICE?: type
Warning: Singleton non-terminal, maybe SPLICE?: vernacular
Warning: doc/sphinx/proof-engine/tactics.rst line 3950: Repeated cmd: 'Print Hint @ident'
Warning: Nonterminals 'tactic_mode' to 'ltac_use_default' not included in .rst files (2)
Warning: Nonterminal dangling_pattern_extension_rule not included in .rst files
Warning: Nonterminal type not included in .rst files
Warning: Nonterminals 'firstorder_rhs' to 'where' not included in .rst files (2)
Warning: Nonterminals 'control_command' to 'tacticals' not included in .rst files (3)
Warning: Nonterminals 'vernac_aux' to 'subprf' not included in .rst files (2)
Warning: Nonterminal finite_token not included in .rst files
Warning: Nonterminals 'scheme_kind' to 'hint_info' not included in .rst files (3)
Warning: Nonterminal strategy_level_or_var not included in .rst files
Warning: Nonterminals 'command' to 'starredidentref' not included in .rst files (6)
Warning: Nonterminal comment not included in .rst files
Warning: Nonterminals 'hint' to 'tacdef_body' not included in .rst files (2)
Warning: Nonterminals 'int_or_id' to 'fun_scheme_arg' not included in .rst files (3)
Warning: Nonterminal hints_path not included in .rst files
Warning: Nonterminals 'simple_tactic' to 'occurrences' not included in .rst files (44)
Warning: Nonterminals 'ltac_expr' to 'match_hyp' not included in .rst files (28)
TOTAL Nonterminals not included = 101
```

Reply:
OK, will take care of this, possibly by pushing to this very PR if you'd like.

Reply:
Yes, please, and thanks!

Reply:
Done.

Reply:
Thanks!